components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/EditorMicroservice.java

```diff
@@ -353,6 +357,59 @@ public Response fileExistsAtWorkspace(String payload) {
         }
     }
 
+    @POST
+    @Path("/workspace/deploy")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public JsonObject deploy(JsonElement element) throws IOException, SiddhiAppsApiHelperException {
```

Comment:
Here you do not need to throw the exception since this is the end of line for backend and you are not handling the exception in the front end. You can catch the specific exception log it and send a appropriate http status code as a bad request